scripts/optimistic-tock.js

```diff
@@ -0,0 +1,79 @@
+const holidays = require("@18f/us-federal-holidays");
+const moment = require("moment-timezone");
+const scheduler = require("node-schedule");
+const utils = require("../utils");
+
+const TOCK_API_URL = process.env.HUBOT_TOCK_API;
+const TOCK_TOKEN = process.env.HUBOT_TOCK_TOKEN;
+
+let util;
+
+let reminder = (robot) => {
+  const message = {
+    username: "Happy Tock",
+    icon_emoji: "happytock",
+    text: "Don't forget to <https://tock.18f.gov|Tock your time>!",
+    as_user: false,
+  };
+
+  reminder = (tz) => async () => {
+    // Get the Slack users in this timezone.
+    const tzUsers = (await util.getSlackUsers()).filter((u) => u.tz === tz);
+
+    // Now get all the folks who have not submitted their current Tock.
+    const truants = await util.tock.get18FTockTruants(moment.tz(tz), 0);
+
+    const tockSlackUsers = (await util.tock.get18FTockSlackUsers())
+      .filter((tockUser) =>
+        tzUsers.some((tzUser) => tzUser.id === tockUser.slack_id)
+      )
+      .filter((tockUser) => truants.some((t) => t.email === tockUser.email));
+
+    tockSlackUsers.forEach(({ slack_id: slackID }) => {
+      robot.messageRoom(slackID, message);
+    });
+  };
+};
+
+const scheduleReminders = async () => {
+  // Westernmost US timezone. Everyone else in the US should be at this time
+  // or later, so this is where we want to begin.
+  const day = moment.tz("Pacific/Samoa");
+
+  // Proceed to the next Friday, then back up if it's a holiday.
+  while (day.format("dddd") !== "Friday") {
```

Comment:
could this function be replaced with

`const day = moment.tz("Pacific/Samoa").day("Friday")` 

and eliminate line 41?

Reply:
I think `.day("Friday")` is constrained to the current calendar week, so on Saturday, it would return the _previous_ Friday rather than the next one. (Trying it with today, for example, `moment().day("Wednesday")` returns yesterday.)

Thinking through this has made me realize that I completely forgot the piece of logic that causes these thingies to get scheduled again, though, so currently they'll only run once, ever, and then never again. 😂

Reply:
perrrrfect!